crates/uv/tests/pip_compile.rs

```diff
@@ -8653,6 +8653,48 @@ fn git_source_missing_tag() -> Result<()> {
     Ok(())
 }
 
+#[test]
+#[cfg(feature = "git")]
+fn warn_missing_constraint() -> Result<()> {
```

Comment:
Is there a test ensuring this warning is not raised without `tool.uv.sources` present?